pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -680,7 +677,7 @@ func (s *DiscoveryServer) loadAssignmentsForClusterIsolated(proxy *model.Proxy,
 
 	// The service was never updated - do the full update
 	s.mutex.RLock()
-	se, f := s.EndpointShardsByService[string(hostname)][svc.Attributes.Namespace]
+	se, f := s.EndpointShardsByService[string(hostname)]
```

Comment:
why? I'm surprised this passes tests but I added this for a reason, I think we need it for the namespace isolation of hostname . otherwise we will get all endpoints from wrong namespaces